src/ddmd/dmodule.d

```diff
@@ -112,6 +112,21 @@ extern (C++) const(char)* lookForSourceFile(const(char)** path, const(char)* fil
     return null;
 }
 
+// function used to call semantic3 on a module's dependencies
+void semantic3OnDependencies(Module m)
+{
+    if (!m)
+        return;
```

Comment:
I think that happens if the code originates as a string (we don't have a published API for that, but we will).